pootle/apps/pootle_translationproject/models.py

```diff
@@ -424,5 +455,144 @@ def scan_projects(sender, instance, created=False, raw=False, **kwargs):
+        """
+        job_wrapper = cls(None, connection)
+        job_wrapper.job = Job(connection=connection)
+        job_wrapper.id = job_wrapper.job.id
+        job_wrapper.func = func
+        job_wrapper.language = language
+        job_wrapper.project = project
+        job_wrapper.options = options
+        job_wrapper.connection = connection
+        job_wrapper.origin = origin
+        job_wrapper.timeout = timeout
+
+        return job_wrapper
+
+    def get_last_job_key(self):
+
+        return POOTLE_SYNC_LAST_JOB_PREFIX + \
+               "%s.%s" % (self.language.code, self.project.code)
+
+    def create_job(self, status=None, depends_on=None):
+        """
+        Creates Job object with given job ID
+        """
+        args = (self.language, self.project)
+        return Job.create(self.func, args=args, kwargs=self.options,
+                          id=self.id, connection=self.connection,
+                          depends_on=depends_on, status=status)
+
+    def save_enqueued(self, pipe):
+        """
+        Preparing job to enqueue. Works via pipeline.
+        Nothing done if WatchError happens while next `pipeline.execute()`.
+        """
+        job = self.create_job(status=JobStatus.QUEUED)
+        job.origin = self.origin
+        job.enqueued_at = utcnow()
+        if job.timeout is None:
+            job.timeout = self.timeout
+        job.save(pipeline=pipe)
+
+    def save_deferred(self, depends_on, pipe):
+        """
+        Preparing job to defer (add as dependent). Works via pipeline.
+        Nothing done if WatchError happens while next `pipeline.execute()`.
+        """
+        job = self.create_job(depends_on=depends_on, status=JobStatus.DEFERRED)
+        job.register_dependency(pipeline=pipe)
+        job.save(pipeline=pipe)
+
+        return job
+
+
+class SyncJobFunctions(object):
+    SYNC = sync_translation_project
+    UPDATE = update_translation_project
+
+
+POOTLE_SYNC_LAST_JOB_PREFIX = "pootle:sync:last:job:"
+
+
+def create_rq_job(language, project, job_func, **options):
+    queue = get_queue('default')
+    queue.connection.sadd(queue.redis_queues_keys, queue.key)
+
+    job_wrapper = JobWrapper.create(job_func,
+                                    language=language,
+                                    project=project,
+                                    options=options,
+                                    connection=queue.connection,
+                                    origin=queue.name,
+                                    timeout=queue.DEFAULT_TIMEOUT)
+
+    last_job_key = job_wrapper.get_last_job_key()
+
+    with queue.connection.pipeline() as pipe:
+        while True:
+            try:
+                pipe.watch(last_job_key)
+                last_job_id = queue.connection.get(last_job_key)
+                depends_on_wrapper = None
+                if last_job_id is not None:
+                    pipe.watch(Job.key_for(last_job_id))
```

Comment:
`last_job_key` is name of Redis key where we keep `id` of last job saved for given TP.
`last_job_id` is a content of `last_job_key` Redis key.
`Job.key_for()` prepares RQ key for a `Job` object.

```python
last_job_key = 'pootle:sync:last:job:fr.tutorial'
last_job_id = redis_conn.get(last_job_key)
```
and  `Job.key_for(last_job_id)` returns `b'rq:job:' + last_job_id.encode('utf-8')`